### PR TITLE
Storage Backend read implementation using Stow

### PR DESCRIPTION
**Pull request by Aayyush**

### CODE REVIEW

File: server/jobs/storage_backend.go [hunk 1]
```diff
@@ -28,8 +29,47 @@ type storageBackend struct {
 	containerName string
 }
 
-func (s *storageBackend) Read(key string) ([]string, error) {
-	return []string{}, nil
+func (s *storageBackend) Read(key string) (logs []string, err error) {
```
Comment (nishkrishnan) [inline server/jobs/storage_backend.go:32]: This function is so weird.  The return result is a variable that's nested within another func which gets passed to another func.  I don't know this lib so I'm trusting that this is the best way to do this.

But at minimum I think we should not used named return values for the outer function and just return logs, error at the bottom.  I generally don't approve of named return values because if the values get updated in different places it makes the code hard to follow.

Comment (Aayyush) [reply server/jobs/storage_backend.go]: Yeah looking at the documentation, there's only two APIs for accessing an item: 
- `stow.WalkContainer()`
- `stow.GetItemByURL()`

We could use the `GetItemByURL()` method but that would mean we would need to have backend specific implementation to get the item URL, which implies having a generic interface and implementations that return the URL for an item, defeating the purpose of using stow for a backend agnostic storage solution. 

In our case, we would need s3 specific object URL generation which would be hardcoded to something like this
`https://s3-%s.amazonaws.com/%s/%s", i.container.Region(), i.container.Name(), i.Name()`

However, `stow.WalkContainer()` is the preferred way to read items from a container. Moreover, we have added failure stats to both the Read and Write methods. So, if something goes wrong, we'll should know about it looking at the dashboard!

Comment (Aayyush) [reply server/jobs/storage_backend.go]: I have removed the named return values!

**Review by msarvar — APPROVED** [review]
